### ISSUE TITLE
Impossible to implement resource providers for Azure

### ISSUE THREAD

**Issue by ThomasArdal** [Azure/azure-resource-provider-sdk#70]
I have been trying for months now, to create a resource provider to show up in the marketplace. I have followed the [example](https://github.com/MetricsHub/AzureStoreRP) given and read through every document on the [onboarding guide](https://github.com/Azure/azure-resource-provider-sdk). When I submit my application for review (publish to staging), I get an error report back on email, saying that some of the expected calls fails.

The problem here is that the failed calls are on URLs that aren't documented anywhere (like https://mywebsite.azurewebsites.net/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/{provider}/applications/resource{someid}?api-version=2016-06-08). For obvious reasons, my API returns a 404 on these calls.

I don't understand how anyone have been able to implement a resource provider, without a Microsoft engineer deeply involved in the implementation. Maybe it's just me that haven't found the right documentation, but similar issues on this repo, makes me think that updated documentation and examples are needed.

**Comment by Dmitry-Me** [comment]
This repo is big time outdated - it documents "old" RP interfaces and you get calls for a "newer" RP interface.

**Comment by ThomasArdal** [comment]
@Dmitry-Me Ok makes sense. Could you point me in the direction of the documentation and examples for the "newer" RP interface?

**Comment by Dmitry-Me** [comment]
I don't know of a publicly available guide for that. Your best bet is to contact support via the marketplace interface.

**Comment by ThomasArdal** [comment]
Thanks.

**Comment by sjkp** [comment]
Did you ever find a way @thomasardal

**Comment by ThomasArdal** [comment]
@sjkp No. The thing is so poorly documented, that the cost benefit simply doesn't add up.

**Comment by sjkp** [comment]
There's actually some updated guidelines here: https://github.com/Azure/azure-resource-manager-rpc
They however dont include code samples, and it is pobably worth getting in touch with Microsoft before venturing out on a long implementation journey
